src/Kiota.Builder/CodeElementOrderComparer.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Kiota.Builder {
+    public class CodeElementOrderComparer : IComparer<CodeElement>
+    {
+        public int Compare(CodeElement x, CodeElement y)
+        {
+            var isXNull = x == null;
+            var isYNull = y == null;
+            if(isXNull && isYNull) return 0;
+            else if (isXNull) return -1;
+            else if(isYNull) return 1;
+
+
+            return GetTypeFactor(x).CompareTo(GetTypeFactor(y)) * typeWeight + 
+                (x.Name?.CompareTo(y.Name) ?? 0) * nameWeight + 
+                GetParametersFactor(x).CompareTo(GetParametersFactor(y)) * parametersWeight;
```

Comment:
nit: switch expression is slightly easier to parse in these situations while reading the code:
```suggestion
            return (x, y) switch
            {
                (null, null) => 0,
                (null, _) => -1,
                (_, null) => 1,
                _ => GetTypeFactor(x).CompareTo(GetTypeFactor(y)) * typeWeight +
                    (x.Name?.CompareTo(y.Name) ?? 0) * nameWeight +
                    GetParametersFactor(x).CompareTo(GetParametersFactor(y)) * parametersWeight;
            };
```

Reply:
oooooh! I didn't that kind of syntax was supported in dotnet! more functional style!